add file

Vlados-XiMiK opened the pull request:
Added a program for searching repositories

Vlados-XiMiK committed:
add file

Added a program for searching repositories

Gmust commented on laba3.php:
```diff
@@ -0,0 +1,64 @@
+<?php
+
+function search_github_repositories($query) {
+    $url = "https://api.github.com/search/repositories";
```
Use an environment variable or constant instead of inserting the string directly

Gmust commented on laba3.php:
```diff
@@ -0,0 +1,64 @@
+    $context = stream_context_create($options);
+    $response = file_get_contents("$url?" . http_build_query($params), false, $context);
+
+    if ($response !== false) {
```
Use !$response instead of $response !== false for better space optimization

Gmust commented on laba3.php:
```diff
@@ -0,0 +1,64 @@
+        $repositories = $data['items'];
+        return $repositories;
+    } else {
+        echo "Під час отримання репозиторіїв сталася помилка.";
```
Also throw  exception with error

Gmust commented on laba3.php:
```diff
@@ -0,0 +1,64 @@
+    echo "Зірки: " . $repository['stargazers_count'] . "\n";
+    echo "Форки: " . $repository['forks_count'] . "\n";
+    echo "Спостерігачі: " . $repository['watchers_count'] . "\n";
+    echo "Мова: " . $repository['language'] . "\n\n";
```
Could be a couple of PLs, fix that

Gmust reviewed (commented):
Please fix this issues and provide me new code for further review. Also, I strongly recommend using other programming languages instead of PHP (because it sucks)
![photo_2024-05-06_10-37-32](https://github.com/Vlados-XiMiK/SearchRepo-GitHub/assets/95234452/debd4f79-9b02-4a71-8743-403d096b5812)

Vlados-XiMiK committed:
update code

Updated the search_github_repositories function to use an environment variable for the GitHub API URL and fixed the output of the repository language. Also added exception handling in case of failed API request.

Gmust reviewed (approved):
Well done, approved
